### DISCUSSION TITLE
Adding table logic to DocBook reader

### DISCUSSION THREAD

**Discussion by lifeunleaded** [jgm/pandoc#10178] [Ideas]
With the introduction of table custom styles in the docx writer, I'd like to add parsing of the `tabstyle` attribute in the DocBook reader. While I'm at it, I'd like to also add parsing of `id` and `class` (which for tables is an XML attribute in the Docbook schema).

I've got working code (changing `parseTable` to use `tableWith` in the DocBook reader), but as per the contribution guidelines, I'm submitting the idea for scrutiny first before adding tests and writing a pull request.

Any opinions?

**Comment by jgm** [comment]
`id` and `class` certainly welcome.  With `tabstyle`, is your idea to populate `custom-style` with this?

**Reply by lifeunleaded** [reply]
@jgm yes, exactly. They serve very similar purposes in an abstract sense, so a user coming from docbook could match existing values for `tabstyle` by creating custom table styles with the same name in the custom reference docx. It seems to work and gives a very pleasant flow without needing a filter.

**Reply by lifeunleaded** [reply]
https://github.com/jgm/pandoc/pull/10183 is in place to show the idea.

**Comment by lifeunleaded** [comment]
Added https://github.com/jgm/pandoc/issues/10181 for this
